crates/common_lang_types/src/text_with_carats.rs

```diff
@@ -116,19 +114,35 @@ fn text_with_carats_and_line_count_buffer_and_line_numbers(
         };
 
         if should_print_carats {
-            let start_of_carats = actual_span.start.saturating_sub(start_of_line as u32);
             let line_len = line_content.len() as u32;
-            let end_of_carats = std::cmp::min(
+            let start_of_carats = actual_span.start.saturating_sub(start_of_line as u32);
```

Comment:
nit: no worries here, but it's slightly easier to review if you don't re-order lines like this. Before submitting PRs, it's a good practice to do `git diff main` and see where there are minor changes like this that add visual noise but don't materially affect the PR. (Likewise, if there are refactors that you can extract, that's good to make as a separate PR, but so far I see none of those.)